src/torrent.cpp

```diff
@@ -5078,33 +5096,33 @@ bool is_downloading_state(int const st)
 		// setting file priority on a torrent that doesn't have metadata yet is
 		// similar to having passed in file priorities through add_torrent_params.
 		// we store the priorities in m_file_priority until we get the metadata
-		file_storage const& fs = m_torrent_file->files();
-		if (index < file_index_t(0) || (valid_metadata() && index >= fs.end_file()))
+		if (index < file_index_t(0)
```

Comment:
Is `file_index_t` type allowed to have negative values? What does it represent with negative value?

Reply:
it depends on what you mean by "allowed". It refers to an invalid index, similar to an index thats >= the number of files

Reply:
Ah, got it.